src/OrchardCore/OrchardCore.Secrets.Abstractions/ISecretService.cs

```diff
@@ -0,0 +1,16 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using OrchardCore.Secrets.Models;
+
+namespace OrchardCore.Secrets;
+
+public interface ISecretService
```

Comment:
No I don't think so, can still be used to create protected data knowing that it can be distributed.

Secrets can be used in place of passwords, api keys, but the main goal is to exchange these secrets that are sensitive data across servers/instances in a secure way in place of having to share passwords and api keys in a non secure way.

From recently we now can deploy users, we could update this deployment to use dedicated RSA encryption and RSA signing, as we do here to deploy secrets, but maybe in a separate PR.

Reply:
But isn't the point here to also offer a way to persist secrets locally (like to the DB) in a safe and reliable way? Because we use `IDataProtectionProvider` a lot for that too while it isn't really suitable for that.

Reply:
As I understood `IDataProtectionProvider` can be still used locally as it is currently.

And this is when we exchange sensitive data, like when deploying secrets, particularly across different servers.

Here too we need the help of @kevinchalet to confirm or not ;)

Reply:
Related: https://github.com/OrchardCMS/OrchardCore/pull/7891#discussion_r1396337070

Reply:
I meant we use Secrets (and then the `SecretService`) to exchange sensitive data (including secrets themselves), for example between servers not using the same database. This by using RSA keys

And as I understood all `IDataProtectionProvider` can be let as is.

Reply:
> Symmetric cryptography typically gets used when speed is the priority over increased security, keeping in mind that encrypting a message **still offers a high level of security**. Some of the most common use cases for symmetric cryptography include: 

>    Banking: Encrypting credit card information or other personally identifiable information (PII) required for transactions 
    Data storage: Encrypting data stored on a device **when that data is not being transferred**

When data need to be deployed/transfered we use additional Asymmetric RSA keys to wrap data that are only encrypted with a symetric key.

Reply:
The new protection we would add is related to th following

> To communicate a symmetric key and IV to a remote party, you usually encrypt the symmetric key by using asymmetric encryption. Sending the key across an insecure network without encryption is unsafe because anyone who intercepts the key and IV can then decrypt your data.